src/java/map/FloatMask.java

```diff
@@ -761,6 +788,48 @@ public FloatMask spike(int radius, BinaryMask limiter) {
         return this;
     }
 
+    public FloatMask addFloatMaskCenteredAtLocationWithSize(FloatMask other, Vector2f location, int size) {
+        if (size > getSize()) {
+            throw new IllegalArgumentException("Added mask size is larger than base mask size");
+        }
+        FloatMask maskToBeAdded = other.copy().setSize(size);
+        add(maskToBeAdded, location, true);
+        VisualDebugger.visualizeMask(this);
+        return this;
+    }
+
+    public FloatMask useBrush(Vector2f location, String brushName, float intensity, int size) {
+        FloatMask brush = loadBrush(brushName, new SymmetrySettings(Symmetry.NONE, Symmetry.NONE, Symmetry.NONE));
+        brush.multiply(intensity / brush.getMax());
+        addFloatMaskCenteredAtLocationWithSize(brush, location, size);
+        VisualDebugger.visualizeMask(this);
+        return this;
+    }
+
+    public FloatMask useBrushRepeatedlyCenteredWithinArea(BinaryMask area, String brushName, int size, int frequency, float intensity) {
+        if (size > getSize()) {
+            throw new IllegalArgumentException("Added mask size is larger than base mask size");
+        }
+        List<Vector2f> possibleLocations = new ArrayList<Vector2f>(area.getAllCoordinatesEqualTo(true, 1));
+        int length = possibleLocations.size();
+        FloatMask brush = loadBrush(brushName, new SymmetrySettings(Symmetry.NONE, Symmetry.NONE, Symmetry.NONE));
+        brush.multiply(intensity / brush.getMax()).setSize(size);
+        for (int z = 0; z < frequency; z++) {
+            add(brush, possibleLocations.get(random.nextInt(length)), true);
```

Comment:
If you dont want it to repeat positions you can use remove instead of get. You would then need to check if any are left as well. I would suggest using a Linked list then

Reply:
I do want it to repeat...